### PR TITLE
Accept const refs instead of const unique_ptr refs in reduce and scan APIs.

### LINKED ISSUE #10393: [FEA] Aggregation APIs shouldn't take aggregation parameters as unique_ptr
**Is your feature request related to a problem? Please describe.**

Some aggregation APIs like `cudf::reduce` take the aggregation parameter as a [`unique_ptr<aggregation> const&`](https://github.com/rapidsai/cudf/blob/1e5b01f1f1e4e97b930de45074dc7109e22a9744/cpp/include/cudf/reduction.hpp#L68).

They should not do this. There is effectively[ _never_ a reason to pass a parameter as `unique_ptr<T> const&` ](https://www.modernescpp.com/index.php/c-core-guidelines-passing-smart-pointer).

**Describe the solution you'd like**

Pass as a `reduce_aggregation const&` instead.

### PR DESCRIPTION
**Pull request by vyasr**
## Description

There is almost never a good reason to pass arguments as `unique_ptr<T> const&`. Since those arguments cannot be modified, the only use case is accessing the underlying pointer, at which point the function better communicates its intent by accepting the underlying pointer/reference as an argument instead and is also more flexible as a result.

Resolves #10393

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by bdice — APPROVED** [review]
Approving C++ and Cython changes.

**Review by ttnghia — APPROVED** [review]

File: cpp/include/cudf/detail/scan.hpp [hunk 1]
```diff
@@ -38,7 +38,7 @@ namespace detail {
  *                           `agg` is not Min or Max.
  *
  * @param input The input column view for the scan.
- * @param agg unique_ptr to aggregation operator applied by the scan.
+ * @param agg aggregation operator applied by the scan.
```
Comment (davidwendt) [inline cpp/include/cudf/detail/scan.hpp:41]:
```suggestion
 * @param agg Aggregation operator applied by the scan
```

File: cpp/include/cudf/detail/scan.hpp [hunk 2]
```diff
@@ -64,7 +64,7 @@ std::unique_ptr<column> scan_exclusive(column_view const& input,
  *                           but the `agg` is not Min or Max.
  *
  * @param input The input column view for the scan.
- * @param agg unique_ptr to aggregation operator applied by the scan.
+ * @param agg aggregation operator applied by the scan.
```
Comment (davidwendt) [inline cpp/include/cudf/detail/scan.hpp:67]:
```suggestion
 * @param agg Aggregation operator applied by the scan
```

File: cpp/include/cudf/detail/scan.hpp (hunk 1 above)
Comment (vyasr) [inline cpp/include/cudf/detail/scan.hpp:41]:
```suggestion
 * @param agg Aggregation operator applied by the scan.
```

File: cpp/include/cudf/detail/scan.hpp [hunk 3]
```diff
@@ -38,7 +38,7 @@ namespace detail {
  *                           `agg` is not Min or Max.
  *
  * @param input The input column view for the scan.
- * @param agg unique_ptr to aggregation operator applied by the scan.
+ * @param agg Aggregation operator applied by the scan.
```
Comment (vyasr) [inline cpp/include/cudf/detail/scan.hpp:41]:
```suggestion
 * @param agg Aggregation operator applied by the scan
```

**Comment by davidwendt** [comment]
Can you add a real description to this PR? It will be added to the changelog at the end of the release.

**Comment by bdice** [comment]
> Can you add a real description to this PR? It will be added to the changelog at the end of the release.

Only the PR title is added to the changelog. The description is used for the merge commit message. (Still needs a real description.)

**Comment by vyasr** [comment]
Added a description

**Review by bdice — APPROVED** [review]

**Comment by vyasr** [comment]
@gpucibot merge

**Pull request merged**